File: src/test/java/pages/geographypage/ReleaseEvaluationPage.java
package pages.geographypage;

import com.microsoft.playwright.Page;
import pages.base.BasePage;

/**
 * Release Evaluation Page - Release Evaluation Approval
 * Screen: Release Evaluation
 */
public class ReleaseEvaluationPage extends BasePage {

    private static final String RELEASE_EVAL_HEADER = "//h3[text()='Release Evaluation']";
    private static final String SAVE_BTN = "//*[text()=' Save ']";
    private static final String APPROVE_BTN = "//button[text()=' Approve ']";
    private static final String CONFIRM_BTN = "//button[text()=' Confirm ']";

    private static final String SUCCESS_MESSAGE = "Release evaluation have been saved successfully.";
    private static final String APPROVED_MESSAGE = "Release evaluation approved. Redirecting to Geography Naming...";

    public ReleaseEvaluationPage(Page page) {
        super(page);
    }

    // ============= Screen Validation =============

    public boolean isReleaseEvaluationScreenDisplayed() {
        return isElementVisible(RELEASE_EVAL_HEADER);
    }

    public boolean isOnReleaseEvaluationScreen() {
        return isOnScreenByURL("release-evaluation");
    }

    // ============= Actions =============

    public void clickSave() {
        clickElement(SAVE_BTN);
    }

    public void clickApprove() {
        clickElement(APPROVE_BTN);
    }

    public void clickConfirm() {
        clickElement(CONFIRM_BTN);
    }

    public void approveReleaseEvaluation() {
        clickApprove();
        clickConfirm();
    }

    // ============= Verification =============

    public boolean verifySaveSuccess() {
        return verifySuccessMessage(SUCCESS_MESSAGE);
    }

    public boolean verifyApprovalSuccess() {
        return verifySuccessMessage(APPROVED_MESSAGE);
    }

    // ============= Profile Data Validation =============

    public void validateProfileData(java.util.Map<String, String> data) {
        // Validates that profile data section is visible on Release Evaluation screen
        validateProfileDataIsVisible(data);
    }
}

File: pages/releaseEvaluationPage.ts
import { Page } from '@playwright/test';
import { BasePage } from './basePage';

export class ReleaseEvaluationPage extends BasePage {
  readonly saveButton;
  readonly approveButton;
  readonly confirmButton;
  readonly header;

  constructor(page: Page) {
    super(page);
    this.saveButton = page.locator("//h3[normalize-space()='Release Evaluation']//following::button[normalize-space()='Save']");
    this.approveButton = page.locator("//h3[normalize-space()='Release Evaluation']//following::button[normalize-space()='Approve']");
    this.confirmButton = page.getByRole('button', { name: /confirm/i });
    this.header = page.locator('//h3[contains(text(),"Release Evaluation")]');
  }

  async isReleaseEvaluationDisplayed(): Promise<boolean> {

    await this.page.waitForURL(/.*release-evaluation/, { timeout: 20000 });  

     return await this.header.isVisible({ timeout: 10000 });
  }

  async clickSave(): Promise<void> { 
    await this.saveButton.waitFor({ state: 'visible', timeout: 10000 }); 
    await this.waitForDataAndClickRefresh();
    await this.saveButton.click(); 
  }

  async clickApprove(): Promise<void> {
    await this.waitForLoaderToDisappear();
    await this.approveButton.waitFor({ state: 'visible', timeout: 10000 });
    await this.approveButton.click();
  }

  async clickConfirm(): Promise<void> {
    await this.waitForLoaderToDisappear();
    await this.confirmButton.waitFor({ state: 'visible', timeout: 10000 });
    await this.confirmButton.click();
  }

  async verifySaveSuccess(message: string): Promise<boolean> {
    const toast = this.page.locator('body').getByText(message, { exact: false });
    await toast.waitFor({ state: 'visible', timeout: 10000 });
    await this.waitForLoaderToDisappear();
    return await toast.isVisible();
  }

  async verifyApprovalSuccess(message: string): Promise<boolean> {
    const toast = this.page.locator('body').getByText(message, { exact: false });
    await toast.waitFor({ state: 'visible', timeout: 10000 });
    await this.waitForLoaderToDisappear();
    return await toast.isVisible();
  }

  async validateProfileData(): Promise<void> {
    await this.page.getByText(/Profile Data/i).first().waitFor({ state: 'visible' });
  }
}

